tests/treeherder/test_api.py

```diff
@@ -20,24 +21,52 @@
 @json_handler
 def handle_rest_api(request, response):
     """Simple JSON handler for the Treeherder Rest API."""
-    filename = '%s.json' % request.request_path.rsplit('/', 2)[1]
+    url_fragments = urlparse.urlparse(request.url)
+    query_options = urlparse.parse_qs(url_fragments.query)
+    api_endpoint = url_fragments.path.rsplit('/', 2)[1]
 
-    with open(os.path.join(HERE, '29258f59e545', filename), 'r') as f:
-        return json.loads(f.read())
+    # Use API endpoint to load reference JSON data
+    with open(os.path.join(HERE, 'data', '%s.json' % api_endpoint), 'r') as f:
+        data = json.loads(f.read())
+
+    # Apply query options for the loaded data
+    for option, values in query_options.iteritems():
+        # JSON data uses int values while query options have strings - try to convert
+        try:
+            value = int(values[0])
+        except:
+            value = values[0]
+
+        if api_endpoint == 'jobs':
+            data['results'] = [entry for entry in data['results']
```

Comment:
Oh, no I'm wrong. We cannot given that the loop over options is one level above. So we would still check other query options for other api endpoints here. If we don't want that we would have to do all the option checks inside differnt api_endpoint sections.